### ISSUE TITLE
Chrome 404s when requesting same url twice in a row

### ISSUE THREAD

**Issue by pdelanauze** [quirkey/sammy#20]
This is some odd behaviour that seems specific to chrome, safari and firefox are unaffected.

I have a request to #/settings/
this request basically toggles a component's visibility and then redirects to #/

This will work the first time
But when requesting this URL the second time, a 404 will be printed and the event won't reach the sammy application.

Now, this only happens when requesting an identical URL a second time without asking for something else in the meanwhile. Example if I ask for #/settings/ and then #/main/ and then #/settings/ everything will work. But if I ask for #/settings/ and then #/settings/ sequentially, the second request will fail with a 404.

**Comment by mrb** [comment]
Can you post some sample code that illustrates this? I use Sammy in Chrome all the time, with redirects, and have never had this issue.

**Comment by pdelanauze** [comment]
```
<a href="#/hello">Test</a>
```

Then in the sammy application: 

```
this.get("#/hello", function(context){
  context.redirect("#/");
});
```

Now if you click on the "test" link the first time, it will be fine, "#/hello" gets triggered and then redirected to #/
Then click on it again, and the request will get stuck in "#/hello" and does not get redirected.

Here's what happens in the logs when doing 2+ clicks on the link element.

```
runRoute get #/
XHR finished loading: "http://localhost/~pat/widget/src/main/webapp/templates/quotes.template".
native hash change exists, using
runRoute get #/hello
Entering hello
runRoute get #/
Exiting hello, should have gotten redirected to #/ now.
XHR finished loading: "http://localhost/~pat/widget/src/main/webapp/templates/quotes.template".
```

**Comment by pengwynn** [comment]
I'm seeing this too, only in chrome

**Comment by mrb** [comment]
Can one of you guys post a Sammy one page app that makes this reproducable? Aaron and I can't get this error to pop up!

**Comment by quirkey** [comment]
last_location was being overriden if you redirected back multiple times. Closed by 66c1fc103b491796e694df25cb1183e1774f1561

All the tests are passing on it, though I'm a little worried its going to break something bigger.

**Comment by pengwynn** [comment]
This indeed fixed the problem I was having in chrome. Thanks!
